Solution3.java

```diff
@@ -0,0 +1,60 @@
+// Time Complexity : O(N)
+// Space Complexity : O(N)
+// Did this code successfully run on Leetcode : Y
+// Three line explanation of solution in plain english
+Iterate on each element in up down fashion. Apply rules when exceeds border
```

Comment:
Good Job! But please mention what your symbols stand for. For example, what is N in this case and refer to the comment on the previous problem to correct your space complexity.